docs/cluster_management.md

```diff
@@ -2,45 +2,76 @@
+```
 
-KUBE_API_VERSIONS env var controls the API versions that are supported in the cluster. The first version in the list is used as the cluster's storage version. Hence, to set a specific version as the storage version, bring it to the front of list of versions in the value of KUBE_API_VERSIONS.
+If you want to have more flexibility for configuring your cluster, you'll need to learn more about the [Salt stack](http://saltstack.com/).
 
-### Switching your config files to a new API version
+## Upgrading a cluster
 
-You can use the kube-version-change utility to convert config files between different API versions.
+GKE handles cluster upgrades for you, so you don't need to worry about. If you already a cluster running in some other environment and want to upgrade Kubernetes version running on it, you can use the `cluster/kube-push.sh` script. It works out of the box if your cluster runs on GCE and most of other providers.
 
+If your cluster is running on GCE and you have a repository with the version of Kubernetes you want to push to your cluster, you just need to do the following:
+```
+make release
+cluster/kube-push.sh
 ```
-$ hack/build-go.sh cmd/kube-version-change
-$ _output/local/go/bin/kube-version-change -i myPod.v1beta3.yaml -o myPod.v1.yaml
+or equivalent
 ```
+hack/dev-build-and-push.sh
+```
+This will build new version of Kubernetes, run basic unit and integration tests on it, and if all tests pass upgrade running Kubernetes to the version you just built.
+
+## Scaling the running cluster
+
+When the Kubelet is given masters address it'll self register in running Kubernetes cluster.
+
+If your cluster runs out short on resources you can easily add more machines to it. If you're using GCE or GKE it is done by resizing Instance Group managing your Nodes. It's done by modifying number of instances on `Compute > Compute Engine > Instance groups > your group > Edit group` page. Instance Group will take care of putting appropriate image on new machines and start them, while Kubelet will register its Node with API server to make it available for scheduling.
+
+In other environments you may need to configure the machine yourself and tell the Kubelet on which machine API server is running.
+
+## Cluster introspection
+
+To learn about the state of the cluster you can use a number of available tools. The simplest one is plain `kubectl` which can `describe node <node_name>`, giving you a summary of all information Kubernetes has about this particular Node. Other available tools include Elasticsearch, Graphana etc. Kubernetes is shipped with a number of so called cluster addons that provide additional functionality, such as monitoring. They run as ordinary pods and can be managed in exactly the same way. More detailed description is available [here](../cluster/addons).
```

Comment:
For things like kubectl commands, it's better to list a full example and be more specific about what they can see there.